src/workingtitle-vcockpits-instruments-cj4/html_ui/Pages/VCockpit/Instruments/Airliners/CJ4/FMC/CJ4_FMC_InitRefIndexPage.js

```diff
@@ -1102,51 +1102,80 @@ class CJ4_FMC_InitRefIndexPage {
         fmc.onNextPage = () => { CJ4_FMC_InitRefIndexPage.ShowPage22(fmc); };
         fmc.updateSideButtonActiveStatus();
     }
+    static ShowPage25ErrorMsg(fmc, msg, rskText) {
+         fmc._templateRenderer.setTemplateRaw([
+            [" ACT[blue]", "ARRIVAL DATA  [blue]"],
+            [" " + msg],
+            [""],
+            [""],
+            [""],
+            [""],
+            [""],
+            [""],
+            [""],
+            [""],
+            [""],
+            ["-----------------------[blue]"],
+            ["<INDEX", rskText + ">"]
+        ]);
+    }
     static ShowPage25(fmc) { //ARR DATA
         fmc.clearDisplay();
-
-        let destinationString = "NO DESTINATION AIRPORT";
-        let approachName = "";
-        let approachFrequency = "";
-        let appRunway = "";
+        
+        let arrAirportText = "";
+        let approachText = "";
+        let rwyThresholdAltText = "";
+        let freqText = "---.--";
+        let gsAngleText = "-.--°";
+        let locTrueBrgText = "---T";
+        
+        let rightSelectionKeyProc = undefined;
 
         let destination = fmc.flightPlanManager.getDestination();
-        if (destination) {
-            destinationString = destination.ident;
-            if (destination.infos) {
-                destinationString = destinationString + "/" + destination.infos.name;
-            }
-        }
-        
-        if (fmc.flightPlanManager.getApproach()) {
+        if (!destination) {
+            CJ4_FMC_InitRefIndexPage.ShowPage25ErrorMsg(fmc, "NO DEST SELECTED", "FPLN");
+            rightSelectionKeyProc = CJ4_FMC_RoutePage.ShowPage1;
+        } else {
             let approach = fmc.flightPlanManager.getApproach();
-            approachName = fmc.flightPlanManager.getApproach().name;
-            approachFrequency = fmc.flightPlanManager.getApproachNavFrequency();
-        }
-        if (fmc.flightPlanManager.getApproachRunway()) {
-            appRunway = fmc.flightPlanManager.getApproachRunway();
-        }
+            if (!approach.name && !fmc.vfrLandingRunway) {
```

Comment:
if approach is null, with this throw an error? Might need to add additional check at startof conditional.

Reply:
approach can never be null. That's why the check is !approach.name instead of !approach:
https://github.com/Working-Title-MSFS-Mods/fspackages/blob/a79139affe944f0df6a2afa538e609258445cde2/src/workingtitle-vcockpits-instruments-airliners/html_ui/Pages/VCockpit/Instruments/Airliners/Shared/WT/FlightPlanManager.js#L1142-L1147